patroni/ha.py

```diff
@@ -432,7 +432,7 @@ def demote(self, delete_leader=True):
             sleep(2)  # Give a time to somebody to take the leader lock
             cluster = self.dcs.get_cluster()
             node_to_follow = self._get_node_to_follow(cluster)
-            self.state_handler.follow(node_to_follow, cluster.leader, recovery=True, need_rewind=True)
+            return self.state_handler.follow(node_to_follow, cluster.leader, recovery=True, need_rewind=True)
```

Comment:
I think we also need to add the same to _do_reinitialize, since it is also called asynchronously, so the leader key waiting logic won't apply anymore (not as import for pg_rewind though).